src/Microsoft.Health.Fhir.Core/Features/Search/CodeSystemResolver.cs

```diff
@@ -0,0 +1,53 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using EnsureThat;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core.Data;
+using Microsoft.Health.Fhir.Core.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search
+{
+    public class CodeSystemResolver : ICodeSystemResolver, IStartable
+    {
+        private readonly IModelInfoProvider _modelInfoProvider;
+        private Dictionary<string, string> _dictionary;
```

Comment:
Should this be initialized to an empty dictionary? If ResolveSystem is somehow run before Start is called it will cause a null ref error.